tools/notary-migration-fix.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# Copyright Project Harbor Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -ex
+
+harbor_db_image=$(docker images goharbor/harbor-db --format "{{.Repository}}:{{.Tag}}")
+harbor_db_path="/data/database"
+
+launch_db() {
+    docker run -d --name fix-notary-migration -v ${harbor_db_path}:/var/lib/postgresql/data ${harbor_db_image} "postgres"
+}
+
+clean_db() {
+    docker stop fix-notary-migration
+    docker rm fix-notary-migration
+}
+
+wait_for_db_ready() {
+    TIMEOUT=12
+    while [ $TIMEOUT -gt 0 ]; do
+        docker exec fix-notary-migration pg_isready | grep "accepting connections"
```

Comment:
If nothing is found, grep will return non-zero value and this script will exit, to avoid this, you need to add ' || true ' at the end.

Reply:
no-zero will still in the while loop tile timeout

Reply:
@wy65701436 did you test it with `set -e`?

Reply:
set -ex means it exits when any command returns non-zero

Reply:
get your points, code updated.